content/webapp/components/IIIFViewer/ViewerTopBar.tsx

```diff
@@ -240,17 +254,20 @@ const ViewerTopBar: FunctionComponent<ViewerTopBarProps> = ({
 
   // We also want to offer download options for each canvas image
   // in the iiif-presentation manifest when it is being viewed.
-  const canvasImageDownloads = mainImageService['@id']
-    ? getDownloadOptionsFromImageUrl({
-        url: mainImageService['@id'],
-        // TODO the dimensions should really be from the painting width and height, not the canvas
-        // it is coincidental that the canvas width and height are the same as the painting width and height
-        // Fix this if/when we remove imageServiceId from the TransformedCanvas
-        // in favour of using the painting property to find the image/image service
-        width: currentCanvas && currentCanvas.width,
-        height: currentCanvas && currentCanvas.height,
-      })
-    : [];
+  const canvasImageDownloads = imageServices
+    .map(imageService => {
+      if (imageService['@id']) {
+        return getDownloadOptionsFromImageUrl({
+          url: imageService['@id'],
+          width: imageService.width || undefined,
+          height: imageService.height || undefined,
+        });
+      } else {
+        return [];
+      }
+    })
+    .flat()
+    .filter(Boolean);
```

Comment:
Why was this filter needed?

Reply:
just removes any undefined's that could have been added to the array